packages/eslint-plugin/rules/i18n-no-flanking-whitespace.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const PROBLEMS_BY_CHAR_CODE = {
+	9: '\\t',
+	10: '\\n',
+	13: '\\r',
+	32: 'whitespace',
+};
+
+function makeFixerFunction( arg ) {
+	return ( fixer ) => {
+		switch ( arg.type ) {
+			case 'TemplateLiteral':
+				return arg.quasis.reduce( ( fixes, quasi ) => {
+					if (
+						'TemplateElement' === quasi.type &&
+						quasi.value.value.match( /^\s|\s$/ )
+					) {
+						fixes.push(
+							fixer.replaceTextRange(
+								[ quasi.start, quasi.end ],
+								`'${ quasi.value.value.trim() }'`
+							)
+						);
+					}
+					return fixes;
+				}, [] );
```

Comment:
This rule doesn't give an error when using template literals (example: ```__( ` Text ` )```). I think it might be related to the way to extract the value from `quasi`, looks like we should use `quasi.value.raw` instead of `quasi.value.value`.

_Referece: https://astexplorer.net/_
<img width="848" alt="Screenshot 2022-02-01 at 11 45 14" src="https://user-images.githubusercontent.com/14905380/151954762-cbd0dd15-12fe-433b-80f4-915bd963b0f9.png">

Reply:
Thanks for linking to the AST, that was really helpful Carlos!

I didn't dig too deep but the few things I tried didn't seem to work, or at least I was unable to get the unit tests to work well with any updates. Not that we absolutely need a unit test for this case. I figure since the rule will still catch the case, it's safe to remove the template literal fixer for now.